### PR TITLE
feat: correct not-realization building

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/Building/CorrectBuildingNotRealizationLambdaHandler.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        protected override TicketError? MapDomainException(DomainException exception, CorrectBuildingNotRealizationLambdaRequest request)
+        {
+            return exception switch
+            {
+                BuildingHasInvalidStatusException => new TicketError(
+                    ValidationErrorMessages.Building.BuildingCannotBeCorrectedFromNotRealizedToPlanned,
+                    ValidationErrorCodes.Building.BuildingCannotBeCorrectedFromNotRealizedToPlanned),
+                BuildingHasInvalidBuildingGeometryMethodException => new TicketError(
```
Comment (ArneD) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/Building/CorrectBuildingNotRealizationLambdaHandler.cs:60]: Is this needed?

File: src/BuildingRegistry.Api.BackOffice/Building/BuildingController-CorrectNotRealization.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            catch (DomainException exception)
+            {
+                throw exception switch
+                {
+                    BuildingIsRemovedException => new ApiException(ValidationErrorMessages.Building.BuildingRemoved, StatusCodes.Status410Gone),
+                    BuildingHasInvalidStatusException => CreateValidationException(
+                        ValidationErrorCodes.Building.BuildingCannotBeCorrectedFromNotRealizedToPlanned,
+                        string.Empty,
+                        ValidationErrorMessages.Building.BuildingCannotBeCorrectedFromNotRealizedToPlanned),
+                    BuildingHasInvalidBuildingGeometryMethodException =>CreateValidationException(
```
Comment (ArneD) [inline src/BuildingRegistry.Api.BackOffice/Building/BuildingController-CorrectNotRealization.cs:113]: Needed?

File: src/BuildingRegistry.Projections.Extract/BuildingUnitExtract/BuildingUnitExtractV2Projections.cs [hunk 3]
```diff
@@ -156,6 +156,16 @@ public BuildingUnitExtractV2Projections(IOptions<ExtractConfig> extractConfig, E
                     }, ct);
             });
 
+            When<Envelope<BuildingUnitWasNotRealizedV2>>(async (context, message, ct) =>
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Extract/BuildingUnitExtract/BuildingUnitExtractV2Projections.cs:159]: good catch

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/BuildingRegistry.Api.BackOffice/Building/BuildingController-CorrectNotRealization.cs (hunk 2 above)
Comment (jvandaal) [reply src/BuildingRegistry.Api.BackOffice/Building/BuildingController-CorrectNotRealization.cs:113]: Yes, see GAWR-3701

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/Building/CorrectBuildingNotRealizationLambdaHandler.cs (hunk 1 above)
Comment (jvandaal) [reply src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/Building/CorrectBuildingNotRealizationLambdaHandler.cs:60]: Yes, see GAWR-3701

**Pull request merged**